### PR TITLE
update set env variables

### PR DESCRIPTION
**Pull request by MarcelWilnicki**
what changed:

I expanded the python functions for searching for libgomp packages. On Jenkins searching from "/" doesn't work as expected so I added try except to resolve that issue. Ideally we could investigate why the process throws an OSError.

in set_env.sh I added a case for Jenkins, for some reason Jenkins has a problem with the syntax in bash, so my solution was to declare a JENKINS variable inside groovy, then a jenkins scenario will be run inside bash with a different syntax, (but doing the same thing). Ideally we could also investigate why this syntax error appears. Perhaps there is some specific version of bash, or something that emulates bash but is not in fact bash? Either way, it was beyond my expertise, perhaps Kuba would know more.

### CODE REVIEW

File: set_env_variables.sh [hunk 1]
```diff
@@ -8,10 +8,19 @@ log() {
   echo -e "${COLOR_CYAN}$1${COLOR_DEFAULT}"
 }
 
-if [ -z ${SCRIPT_DIR+x} ]; then
-  SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+# if the script is run with Jenkins, then set the SCRIPT_DIR manually, otherwise use the old way (SET JENKINS TO 1)
+if [[ -z "${JENKINS}" ]]; then
```
Comment (3sky) [inline set_env_variables.sh:12]: What's the source of this variable ?

File: utils/setup/gen_ld_preload.py [hunk 2]
```diff
@@ -11,12 +11,30 @@
...
 
-for path in Path("/").rglob("libgomp*"):
-    if ".so" in path.name:
+try:
+    for path in Path("/").rglob("libgomp*"):
+        if ".so" in path.name:
+            ld_preload.append(str(path))
+except OSError as e:
+    print(e)
+    for path in Path("/lib").rglob("libgomp*"):
```
Comment (3sky) [inline utils/setup/gen_ld_preload.py:20]: What's the reason behind this strange "error-handling"?

File: set_env_variables.sh (hunk 1 above)
Comment (MarcelWilnicki) [reply set_env_variables.sh:12]: it needs to be set in vigilan sniffle before runing this script, something like "export JENKINS=1"

File: utils/setup/gen_ld_preload.py (hunk 2 above)
Comment (MarcelWilnicki) [reply utils/setup/gen_ld_preload.py:20]: when I search from root when run in Jenkins this is an error which python throws. However when I run from anywhere below like "/lib" it works as usual.

File: set_env_variables.sh (hunk 1 above)
Comment (3sky) [reply set_env_variables.sh:12]: I would recommend using 'IS_JENKINS' var name. It describes situation more clear 🙂

File: utils/setup/gen_ld_preload.py (hunk 2 above)
Comment (3sky) [reply utils/setup/gen_ld_preload.py:20]: Jenkins isn't allowed to run from /root directly on agent. Try use path like `/root/miniforge3/envs` maybe it will works better.

Comment (MarcelWilnicki) [reply utils/setup/gen_ld_preload.py:20]: I see, I will change it for "/root" however other packages are located in "/lib"

File: set_env_variables.sh [hunk 3]
```diff
@@ -8,10 +8,19 @@ log() {
...
+  # not run with JENKINS
+  if [ -z ${SCRIPT_DIR+x} ]; then
+  SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+  fi
+else
+  # run with JENKINS, need to set the SCRIPT_DIR in a different way, more manually
+  SCRIPT_DIR=`pwd`
+  SCRIPT_DIR=$SCRIPT_DIR/ampere_model_library
```
Comment (jan-grzybek-ampere) [inline set_env_variables.sh:20]: Why not explicitly SCRIPT_DIR="pwd/ampere_model_library"?

File: set_env_variables.sh [hunk 4]
```diff
@@ -8,10 +8,19 @@ log() {
   echo -e "${COLOR_CYAN}$1${COLOR_DEFAULT}"
 }
 
-if [ -z ${SCRIPT_DIR+x} ]; then
-  SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+# if the script is run with Jenkins, then set the SCRIPT_DIR manually, otherwise use the old way (SET JENKINS TO 1)
+if [[ -z "${JENKINS}" ]]; then
+  # not run with JENKINS
+  if [ -z ${SCRIPT_DIR+x} ]; then
+  SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
```
Comment (jan-grzybek-ampere) [inline set_env_variables.sh:15]: Indent missing

File: set_env_variables.sh (hunk 1 above)
Comment (jan-grzybek-ampere) [reply set_env_variables.sh:12]: I suggest following logic:
 

```
  if IS_JENKINS:
      SCRIPT_DIR="smth"
  elif not SCRIPT_DIR:
      SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
```

File: set_env_variables.sh [hunk 5]
```diff
@@ -34,10 +43,10 @@ sleep 1
 ARCH=$( uname -m )
 if [ "${ARCH}" = "aarch64" ]; then
    python3 $SCRIPT_DIR/utils/setup/gen_ld_preload.py
-   LD_PRELOAD=$(<$SCRIPT_DIR/utils/setup/.ld_preload)
-   export LD_PRELOAD=$LD_PRELOAD
+   LD_PRELOAD=`cat $SCRIPT_DIR/utils/setup/.ld_preload`
    echo "LD_PRELOAD=$LD_PRELOAD"
 fi
+export LD_PRELOAD=$LD_PRELOAD
```
Comment (jan-grzybek-ampere) [inline set_env_variables.sh:49]: This will fail on x86 because LD_PRELOAD will be undefined variable now.

File: set_env_variables.sh [hunk 6]
```diff
@@ -34,10 +43,10 @@ sleep 1
 ARCH=$( uname -m )
 if [ "${ARCH}" = "aarch64" ]; then
    python3 $SCRIPT_DIR/utils/setup/gen_ld_preload.py
-   LD_PRELOAD=$(<$SCRIPT_DIR/utils/setup/.ld_preload)
-   export LD_PRELOAD=$LD_PRELOAD
+   LD_PRELOAD=`cat $SCRIPT_DIR/utils/setup/.ld_preload`
```
Comment (jan-grzybek-ampere) [inline set_env_variables.sh:46]: What's the rationale of this change?

File: utils/setup/gen_ld_preload.py [hunk 7]
```diff
@@ -11,12 +11,30 @@
...
+        ld_preload.append(str(path))
+except OSError as e:
+    print(e)
+    for path in Path("/lib").rglob("libGLdispatch.so.0"):
         ld_preload.append(str(path))
 
-for path in Path("/").rglob("libGLdispatch.so.0"):
-    ld_preload.append(str(path))
+try:
+    for path in Path("/root").rglob("libgomp*"):
```
Comment (jan-grzybek-ampere) [inline utils/setup/gen_ld_preload.py:33]: Let's move that above, where the search for libgomp occurs for the first time

File: set_env_variables.sh (hunk 6 above)
Comment (MarcelWilnicki) [reply set_env_variables.sh:46]: LD_PRELOAD=$(<$SCRIPT_DIR/utils/setup/.ld_preload) for some reason doesn't work in jenkins

LD_PRELOAD=`cat $SCRIPT_DIR/utils/setup/.ld_preload` does exactly the same as above, and additionaly I think it is a bit more clear

File: set_env_variables.sh (hunk 5 above)
Comment (MarcelWilnicki) [reply set_env_variables.sh:58]: actually, why do we need this condition? Wouldn't it be the same for all platforms?

File: set_env_variables.sh [hunk 8]
```diff
@@ -8,8 +8,22 @@ log() {
...
+#    SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+#  fi
+#else
+  # run with JENKINS, need to set the SCRIPT_DIR in a different way, more manually
+#  SCRIPT_DIR=`pwd`/ampere_model_library
+#fi
+
+# checks if IS_JENKINS has length equal to zero
+if [[ -z "${IS_JENKINS}" ]]; then
+  SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
```
Comment (jan-grzybek-ampere) [inline set_env_variables.sh:24]: What if it's not "jenkins", but script_dir is set (e.g. by model_zoo_dev)?

File: set_env_variables.sh (hunk 5 above)
Comment (jan-grzybek-ampere) [reply set_env_variables.sh:58]: No, we dont need LD_PRELOAD hack on x86

File: set_env_variables.sh (hunk 6 above)
Comment (jan-grzybek-ampere) [reply set_env_variables.sh:51]: Got it.

Comment (jan-grzybek-ampere) [reply set_env_variables.sh:51]: Actually given that CI uses model_zoo_dev, instead of IS_JENKINS variable, this should be solved by passing a proper value (I assume pwd/ampere_model_library) from the level of model_zoo_dev wrapper (setup.sh -> https://github.com/AmpereComputingAI/model_zoo_dev/blob/main/setup.sh). That's the reason SCRIPT_DIR variable was here in the first place.

File: set_env_variables.sh (hunk 8 above)
Comment (jan-grzybek-ampere) [reply set_env_variables.sh:24]: Actually given that CI uses model_zoo_dev, instead of IS_JENKINS variable, this should be solved by passing a proper value (I assume pwd/ampere_model_library) from the level of model_zoo_dev wrapper (setup.sh -> https://github.com/AmpereComputingAI/model_zoo_dev/blob/main/setup.sh). That's the reason SCRIPT_DIR variable was here in the first place.

File: set_env_variables.sh (hunk 5 above)
Comment (MarcelWilnicki) [reply set_env_variables.sh:58]: so this will not fail, it will do nothing, so it could remain as is

**Review by jan-grzybek-ampere — APPROVED** [review]
